sycl/plugins/cuda/CMakeLists.txt

```
@@ -22,22 +22,21 @@ add_library(pi_cuda SHARED
   "pi_cuda.cpp"
 )
 
-add_dependencies(pi_cuda
-  ocl-headers
-)
-
 add_dependencies(sycl-toolchain pi_cuda)
 
 set_target_properties(pi_cuda PROPERTIES LINKER_LANGUAGE CXX)
 
 target_include_directories(pi_cuda
   PRIVATE
     ${sycl_inc_dir}
+    ${OPENCL_ICD_LOADER_HEADERS_DIR}
   PUBLIC
     ${CUDA_INCLUDE_DIRS}
 )
 
-target_link_libraries(pi_cuda PUBLIC OpenCL-Headers cudadrv)
+target_link_libraries(pi_cuda
+  PRIVATE OpenCL-Headers
+  PUBLIC cudadrv)
```

Does any target link with libpi_cuda.so? If not, we can keep this dependency private.
```suggestion
 cudadrv)
```

---

Will try and see. Thx.

---

Done. Required some cleanup as well in unittests.